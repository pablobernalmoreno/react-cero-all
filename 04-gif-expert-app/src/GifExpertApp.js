import React, { useState } from "react";
import PropTypes from "prop-types";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((categorie) => (
          <GifGrid key={categorie} category={categorie} />
        ))}
      </ol>
    </>
  );
};

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.array,
};
