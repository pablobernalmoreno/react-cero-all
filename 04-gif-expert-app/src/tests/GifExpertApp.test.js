import React from "react";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  const categories = ["One Piece", "Kenichi"];
  const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

  test("debe mostrar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
