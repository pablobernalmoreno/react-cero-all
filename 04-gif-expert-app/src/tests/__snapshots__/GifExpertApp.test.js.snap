// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe mostrar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Kenichi"
      key="Kenichi"
    />
  </ol>
</Fragment>
`;
